### PR TITLE
update directory and fill in missing files

### PR DESCRIPTION
**Pull request by jacoblogan**
#### Description of changes:
https://pagemigrationupdates-updates.dmhjrabi3pkql.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/directory/generateDirectory.mjs [hunk 1]
```diff
@@ -23,7 +23,8 @@ async function getMetaStringObj(filePath) {
     try {
       // Using JSON5 because the meta object is a "relaxed" JSON
       // JSON5 can parse the meta object without needing quotes around the object keys
-      const result = JSON5.parse(match[1]);
+      let metaObj = match[1].replaceAll('`',"'");
```
Comment (jacoblogan) [inline src/directory/generateDirectory.mjs:26]: replacing template strings to help parsing the meta object but this still breaks with nested ticks

File: src/pages/[platform]/build-a-backend/graphqlapi/best-practice/index.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+import { getChildPageNodes } from '@/utils/getChildPageNodes';
+import directory from 'src/directory/directory.json';
+
+export const meta = {
+    title: 'Best Practice',
+    description: 'Best Practice',
+    platforms: [
```
Comment (jacoblogan) [inline src/pages/[platform]/build-a-backend/graphqlapi/best-practice/index.mdx:9]: Just used a script to generate these files, the title and route are simple enough to get based on the file path this platforms list however is always just using all platforms.  How can we determine this platform list for one of these "Summary" pages?

File: src/directory/directory.mjs [hunk 3]
```diff
@@ -2,210 +2,1809 @@
  * @type {import('./directory').PageNode}
  */
 export const directory = {
```
Comment (jacoblogan) [inline src/directory/directory.mjs:4]: used a script to fill in directory.mjs, this does not maintain the order that was previously established and instead just recursively iterated over the files that existed.  Will probably want to regenerate with ordering in mind.

File: src/pages/gen2/build-a-backend/auth/index.mdx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { getChildPageNodes } from '@/utils/getChildPageNodes';
+import directory from 'src/directory/directory.json';
+
+export const meta = {
+    title: 'Auth',
+    description: 'Auth',
+    route: '/gen2/build-a-backend/auth'
+};
+
+# Auth
```
Comment (jacoblogan) [inline src/pages/gen2/build-a-backend/auth/index.mdx:12]: wasn't sure what to do with these gen2 summary pages

File: tasks/generate-sitemap.mjs [hunk 5]
```diff
@@ -70,6 +70,9 @@ const createRoutePathList = (directory) => {
       `src/pages${route}/index.mdx`,
       `src/pages${route}/index.tsx`
     ]);
+    if(!route){
```
Comment (jacoblogan) [inline tasks/generate-sitemap.mjs:73]: these changes are to accomodate the currently empty gen2 files

**Review by hbuchel — APPROVED** [review]

**Pull request merged**